Fix typing for Twisted 24.10.0

This should be easy but we need this to be included in 2.12 so I've created this for visibility.

> scrapy/pipelines/media.py:209: error: Cannot assign to a method  [method-assign]

`Failure.stack` is a special deprecated property thing in Twisted 24.10.0, and assigning to it is a no-op, maybe it's better to put the assignment under a Twisted version check instead of e.g. specifically silencing the typing warning .

--- a/scrapy/pipelines/media.py
+++ b/scrapy/pipelines/media.py
@@ -16,9 +16,9 @@ from typing import (
 )
 
 from twisted import version as twisted_version
-from twisted.python.versions import Version
 from twisted.internet.defer import Deferred, DeferredList
 from twisted.python.failure import Failure
+from twisted.python.versions import Version
 
 from scrapy.http.request import NO_CALLBACK, Request
 from scrapy.settings import Settings
@@ -209,7 +209,7 @@ class MediaPipeline(ABC):
             result.cleanFailure()
             result.frames = []
             if twisted_version <= Version("twisted", 24, 10, 0):
-                result.stack = []   # type: ignore[method-assign]
+                result.stack = []  # type: ignore[method-assign]
             # This code fixes a memory leak by avoiding to keep references to
             # the Request and Response objects on the Media Pipeline cache.
             #

--- a/scrapy/pipelines/media.py
+++ b/scrapy/pipelines/media.py
@@ -15,6 +15,8 @@ from typing import (
     cast,
 )
 
+from twisted import version as twisted_version
+from twisted.python.versions import Version
 from twisted.internet.defer import Deferred, DeferredList
 from twisted.python.failure import Failure
 
@@ -206,8 +208,8 @@ class MediaPipeline(ABC):
             # minimize cached information for failure
             result.cleanFailure()
             result.frames = []
-            result.stack = []
-
+            if twisted_version <= Version("twisted", 24, 10, 0):
+                result.stack = []   # type: ignore[method-assign]
             # This code fixes a memory leak by avoiding to keep references to
             # the Request and Response objects on the Media Pipeline cache.
             #

